src/layer.rs

```diff
@@ -458,8 +460,10 @@ impl Layer {
     ) {
         let glyph = glyph.into();
         if !self.contents.contains_key(&glyph.name) {
-            let path = crate::util::default_file_name_for_glyph_name(&glyph.name);
-            self.contents.insert(glyph.name.clone(), path.into());
+            let existing_paths: HashSet<String> =
+                self.contents.values().map(|p| p.to_string_lossy().to_lowercase()).collect();
+            let path = crate::util::default_file_name_for_glyph_name(&glyph.name, &existing_paths);
+            self.contents.insert(glyph.name.clone(), path);
```

Comment:
it feels really bad to be creating this collection anew every time a glyph is inserted; think about the situation where you're making a layer and adding a bunch of glyphs, you'll be rebuilding this set each time?

I would prefer to just maintain a set of existing paths or something?

Reply:
I implemented a HashSet cache for LayerSet and Layer now. Please have a look.